e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java

```diff
@@ -65,6 +66,10 @@ public void When_ExtractRequestReceived_Expect_ExtractStatusAndDocumentDataAdded
 
         var ehrContinue = (Document) waitFor(() -> Mongo.findEhrExtractStatus(conversationId).get(EHR_CONTINUE));
         assertThatExtractContinueMessageWasSent(ehrContinue);
+
+        TimeUnit.SECONDS.sleep(10);
+        var ehrDocument = (Document) waitFor(() -> Mongo.findEhrExtractStatus(conversationId).get(GPC_ACCESS_DOCUMENT));
```

Comment:
Await here pointless because of the line above, need a better way of doing this any ideas?

Problem I had Is the gpcAccessDocument already exists in the db record, nested inside an array element we update data by adding a "sentToMhs" object with some data inside.

Reply:
> Await here pointless because of the line above, need a better way of doing this any ideas?

Should we get rid of `sleep` maybe? `waitFor`, I expect, will end as soon as the condition is met, while `sleep` will always take the whole 10 seconds.

Reply:
My problem is I cant get the condition correct for it to wait, so for example:

all other conditions above have relied on root elements of the DB record, however this one is nested inside if that makes sense? Might be quick to show you if not

Reply:
Maybe we could extend the supplier in `waitFor` so that this:
```
var ehrDocument = (Document) waitFor(() -> Mongo.findEhrExtractStatus(conversationId).get(GPC_ACCESS_DOCUMENT))
```
would become sth similar to this:
```
var document = (Document) waitFor(() -> {
    var ehrDocument = Mongo.findEhrExtractStatus(conversationId).get(GPC_ACCESS_DOCUMENT);
    return getFirstDocumentIfItHasObjectNameOrElseNull(ehrDocument);
})
```
?